src/users/repositories/users.repository.ts

```diff
@@ -3,43 +3,79 @@ import { PrismaService } from 'src/prisma/prisma.service';
 import { CreateUserDto } from '../dto/create-user.dto';
 import { UpdateUserDto } from '../dto/update-user.dto';
 import { UserEntity } from '../entities/user.entity';
+import { isPrismaError } from 'src/common/filters/http-exception/errors/utils/is-prisma-error.util';
+import { handleDatabaseErrors } from 'src/common/filters/http-exception/errors/utils/handle-database-errors.utils';
 
 @Injectable()
 export class UsersRepository {
   constructor(private readonly prisma: PrismaService) {}
 
   async create(createUserDto: CreateUserDto): Promise<UserEntity> {
-    return await this.prisma.user.create({
-      data: createUserDto,
-    });
+    try {
+      return await this.prisma.user.create({
+        data: createUserDto,
+        include: {
+          posts: {
+            select: {
+              title: true,
+              createdAt: true,
+            },
+          },
+        },
+      });
+    } catch (error) {
+      if (isPrismaError(error)) {
+        throw handleDatabaseErrors(error);
+      }
+      throw error;
+    }
   }
 
   async findAll(): Promise<UserEntity[]> {
-    return await this.prisma.user.findMany();
+    return await this.prisma.user.findMany({
```

Comment:
Return await is redundant in this case

```suggestion
    return this.prisma.user.findMany({
```